src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2106,11 +2106,10 @@ public DynamicObject toS(DynamicObject string) {
         }
 
         @Specialization(guards = "isStringSubclass(string)")
-        public Object toSOnSubclass(
-                VirtualFrame frame,
-                DynamicObject string,
-                @Cached("new()") SnippetNode snippetNode) {
-            return snippetNode.execute(frame, "''.replace(self)", "self", string);
+        public Object toSOnSubclass(DynamicObject string,
```

Comment:
Out of scope, but it might make sense to move all of this to Ruby.

Reply:
True, would be nice.